ndcube/utils/wcs.py

```diff
@@ -140,75 +147,144 @@ def _wcs_slicer(wcs, missing_axis, item):
+        for i, _bool in enumerate(missing_axis):
             if not _bool:
                 if index is not len(item):
-                    item_checked.append(item[index])
+                    if isinstance(item[index], (int, np.int64)):
+                        item_checked.append(slice(item[index], item[index]+1))
+                    else:
+                        item_checked.append(item[index])
                     index += 1
                 else:
                     item_checked.append(slice(None, None, None))
             else:
                 item_checked.append(slice(0, 1))
-        # if all are slice in the item tuple
+        # if all are slices in the item tuple
         if _all_slice(item_checked):
-            new_wcs = wcs.slice((item_checked))
+            item_ = (item_checked)
         # if all are not slices some of them are int then
         else:
             # this will make all the item in item_checked as slice.
             item_ = _slice_list(item_checked)
-            new_wcs = wcs.slice(item_)
-            for i, it in enumerate(item_checked):
-                if isinstance(it, int):
-                    missing_axis[i] = True
+    else:
+        raise TypeError("item type is {0}.  ".format(type(item)) +
+                        "Must be int, slice, or tuple of ints and/or slices.")
     # returning the reverse list of missing axis as in the item here was reverse of
     # what was inputed so we had a reverse missing_axis.
-    return new_wcs, missing_axis[::-1]
+    dropped_coords = {}  # Initiating new list to collect dropped coords in the process of slicing.
+    # Checking item_ slices for dropped axes if any.
+    for i, slice_element in enumerate(item_):
+        if missing_axis[i] is False:
+            # Determine the start index.
+            if slice_element.start is None:
+                slice_start = 0
+            else:
+                slice_start = slice_element.start
+            # Determine the stop index.
+            if slice_element.stop is None:
+                # wcs._pixel_shape is a list of the length of each axis.
+                slice_stop = wcs.pixel_shape[i]
+            else:
+                slice_stop = slice_element.stop
+            # Determine the slice's step.
+            # (We will use this is a later version of this code to be more thorough.
+            # For now we'll calculate it and not use it.)
+            # if slice_element.step is None:
+                # slice_step = 1
+            # else:
+                # slice_step = slice_element.step
+            if slice_stop - slice_start == 1:
+                # Set up a list of pixel coords as input to all_pix2world.
+                pix_coords = [0] * len(item_)
+                # Enter pixel coordinate for this axis.
+                pix_coords[i] = slice_element.start
+                # Get real world coordinates of i-th axis.
+                real_world_coords = wcs.all_pix2world(*pix_coords, 0)[i]
```

Comment:
```real_world_coords``` here should be made a single value ```Quantity```.  See [here](https://github.com/sunpy/ndcube/blob/master/ndcube/ndcube.py#L270) for how it's done in ```pixel_to_world```.

Reply:
Yup, changed... Would you like to double check my work @DanRyanIrish? More specifically, I modified the code to the following:
```
# Get real world coordinates of i-th axis.
temp_real_world_coords = wcs.all_pix2world(*pix_coords, 0)[i]
real_world_coords = Quantity(temp_real_world_coords, unit=wcs.wcs.cunit[i])
```
So when I followed the steps on the website to initialize a cube and then slice it with `my_2d_cube = my_cube[0, 0, 0:5]`, I would get:
```
{'em.wl': {'axis': None, 'value': <Quantity 1.02e-09 m>},
 'custom:pos.helioprojective.lat': {'axis': None,
  'value': <Quantity 1.26915033e-05 deg>}}
```
Is this the desired result in its proper form?

Reply:
Hi @kakirastern.  Yes, this no looks right.  Can you also confirm the output of
```python
my_2d_cube = my_cube[0, 0, 0:5]
my_2d_cube._extra_coods_wcs_axis
```

Reply:
The output is `{'em.wl': {'wcs axis': 0, 'value': <Quantity 1.02e-09 m>},
 'custom:pos.helioprojective.lat': {'wcs axis': 1,
  'value': <Quantity 1.26915033e-05 deg>}}`... 

Looks alright to you?

Reply:
And there is some issues regarding the existing tests we will need to work on passing... Could you provide me with some hints or suggestions on how to approach these?